aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,496 @@
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	input := &batch.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: aws.String(computeEnvironmentName),
+		ComputeResources:   &batch.ComputeResourceUpdate{},
+	}
+
+	if d.HasChange("service_role") {
+		input.ServiceRole = aws.String(d.Get("service_role").(string))
+	}
+	if d.HasChange("state") {
+		input.ServiceRole = aws.String(d.Get("state").(string))
+	}
+
+	if d.HasChange("compute_resources") {
+		computeResources := d.Get("compute_resources").([]interface{})
+		if len(computeResources) == 0 {
+			return fmt.Errorf("One compute environment is expected, but no compute environments are set")
+		}
+		computeResource := computeResources[0].(map[string]interface{})
+
+		input.ComputeResources.DesiredvCpus = aws.Int64(int64(computeResource["desired_vcpus"].(int)))
+		input.ComputeResources.MaxvCpus = aws.Int64(int64(computeResource["max_vcpus"].(int)))
+		input.ComputeResources.MinvCpus = aws.Int64(int64(computeResource["min_vcpus"].(int)))
+	}
+
+	log.Printf("[DEBUG] Update compute environment %s.\n", input)
+
+	if _, err := conn.UpdateComputeEnvironment(input); err != nil {
+		return err
+	}
+
+	return resourceAwsBatchComputeEnvironmentRead(d, meta)
+}
+
+func resourceAwsBatchComputeEnvironmentStatusRefreshFunc(d *schema.ResourceData, meta interface{}) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		conn := meta.(*AWSClient).batchconn
+
+		computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+		result, err := conn.DescribeComputeEnvironments(&batch.DescribeComputeEnvironmentsInput{
+			ComputeEnvironments: []*string{
+				aws.String(computeEnvironmentName),
+			},
+		})
+		if err != nil {
+			return nil, FAILED, err
+		}
+
+		if len(result.ComputeEnvironments) == 0 {
+			return nil, FAILED, fmt.Errorf("One compute environment is expected, but AWS return no compute environment")
+		}
+
+		computeEnvironment := result.ComputeEnvironments[0]
+		return result, *(computeEnvironment.Status), nil
+	}
+}
+
+func resourceAwsBatchComputeEnvironmentDeleteRefreshFunc(d *schema.ResourceData, meta interface{}) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		conn := meta.(*AWSClient).batchconn
+
+		computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+		result, err := conn.DescribeComputeEnvironments(&batch.DescribeComputeEnvironmentsInput{
+			ComputeEnvironments: []*string{
+				aws.String(computeEnvironmentName),
+			},
+		})
+		if err != nil {
+			return nil, FAILED, err
+		}
+
+		if len(result.ComputeEnvironments) == 0 {
+			return result, DELETED, nil
+		}
+
+		computeEnvironment := result.ComputeEnvironments[0]
+		return result, *(computeEnvironment.Status), nil
+	}
+}
+
+func isCorrentComputeEnvironmentName(i interface{}, k string) (s []string, es []error) {
```

Comment:
I think there is a typo in Corrent 😄 
Also, it would be better to put this function in validators.go, and rename it to `validateBatchComputeEnvironmentName`